drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,528 @@
+ * @retval  -ENOTSUP if an invalid channel is given
+ *
+ */
+static int hx711_channel_get(struct device *dev,
+							 enum sensor_channel chan,
+							 struct sensor_value *val)
+{
+	struct hx711_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_WEIGHT:
+	case SENSOR_CHAN_ALL:
+		val->val1 = sensor_value_to_double(&data->slope) * (data->reading - data->offset);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+/**
+ * @brief Initialise HX711.
+ *
+ * @param dev Pointer to the hx711 device structure
+ *
+ * @retval 0 on success
+ * @retval -ENOTSUP if an invalid channel is given
+ *
+ */
+static int hx711_init(struct device *dev)
+{
+	LOG_DBG("Initialising HX711\n");
+
+	int ret = 0;
+	struct hx711_data *data = dev->driver_data;
+	const struct hx711_config *cfg = dev->config->config_info;
+
+	LOG_DBG("SCK GPIO port : %s\n", cfg->sck_ctrl);
+	LOG_DBG("SCK Pin : %d\n", cfg->sck_pin);
+	LOG_DBG("DOUT GPIO port : %s\n", cfg->dout_ctrl);
+	LOG_DBG("DOUT Pin : %d\n", cfg->dout_pin);
+	LOG_DBG("RATE GPIO port : %s\n", cfg->sck_ctrl);
+	LOG_DBG("RATE Pin : %d\n", cfg->sck_pin);
+	LOG_DBG("Gain : %d\n", data->gain);
+	LOG_DBG("Offset : %d\n", data->offset);
+	LOG_DBG("Slope : %d.%d\n", data->slope.val1, data->slope.val2);
+
+	/* Configure SCK as output, LOW */
+	data->sck_gpio = device_get_binding(cfg->sck_ctrl);
+	if (data->sck_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->sck_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("SCK pin controller is %p, name is %s\n",
+			data->sck_gpio,
+			data->sck_gpio->config->name);
+
+	ret = gpio_pin_configure(data->sck_gpio,
+							 cfg->sck_pin,
+							 GPIO_OUTPUT_INACTIVE | cfg->sck_flags);
+	if (ret != 0) {
+		return ret;
+	}
+
+#if DT_NODE_HAS_PROP(HX711_NODE, rate_gpios)
+	/* Configure RATE as output, LOW */
+	data->rate_gpio = device_get_binding(cfg->rate_ctrl);
+	if (data->rate_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->rate_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("RATE pin controller is %p, name is %s\n",
+			data->rate_gpio,
+			data->rate_gpio->config->name);
+	ret = gpio_pin_configure(data->rate_gpio,
+			cfg->rate_pin,
+			GPIO_OUTPUT_INACTIVE | cfg->rate_flags);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_set(data->rate_gpio, hx711_config.rate_pin, data->rate);
+	if (ret != 0) {
+		return ret;
+	}
+#endif
+
+	/* Configure DOUT as input */
+	data->dout_gpio = device_get_binding(cfg->dout_ctrl);
+	if (data->dout_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->dout_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("DOUT pin controller is %p, name is %s\n",
+			data->sck_gpio,
+			data->sck_gpio->config->name);
+	ret = gpio_pin_configure(data->sck_gpio,
+							 cfg->dout_pin,
+							 GPIO_INPUT | cfg->dout_flags);
+	if (ret != 0) {
+		return ret;
+	}
+	LOG_DBG("Set DOUT pin : %d\n", cfg->dout_pin);
+
+
+	/* Get a reading to set GAIN */
+	ret = hx711_sample_fetch(dev, SENSOR_CHAN_WEIGHT);
+	if (ret != 0) {
+		return ret;
+	}
```

Comment:
You can remove this block since you always return `ret` below

Reply:
Done.